src/makeCalculator.js

```diff
@@ -4,7 +4,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(currentValue, value) {
+      return currentValue + value;
+    },
+
+    subtract(currentValue, value) {
+      return currentValue - value;
+    },
+
+    multiply(currentValue, value) {
+      return currentValue * value;
+    },
+
+    divide(currentValue, value) {
+      return currentValue / value;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, value) {
+      this.result = callback(this.result, value);
```

Comment:
The `operate` method is expected to directly modify the `result` property. There is no need to pass `currentValue` as the `callback` functions already have access to `this.result`. Simplify the callback functions by removing the `currentValue` parameter and directly operating on `this.result`.

Reply:
fix this comment